test/cpp/end2end/nonblocking_test.cc

```diff
@@ -74,11 +74,15 @@ class NonblockingTest : public ::testing::Test {
   }
 
   bool LoopForTag(void** tag, bool* ok) {
+    intptr_t orig_val = gpr_tls_get(&g_is_nonblocking_test);
+    gpr_tls_set(&g_is_nonblocking_test, static_cast<intptr_t>(1));
```

Comment:
Thanks for looking at it Vijay. Could you clarify the change by maybe adding some comments in the test itself? Like I still don't completely understand why `g_is_nonblocking_test` should be set to `0` in general but to `1` when calling `cq_next`. Does it mean there could be some blocking ops when starting the call, and `cq_next` should not be blocking at all? Why? Some comments might be helpful for me to review and others too.

Reply:
Thanks for the suggestion. I added a bunch of comments and renamed the variable to `g_is_nonblocking_poll` so that it's clear that it only applies to this poll and not the whole test.